src/main/java/com/regnosys/rosetta/common/serialisation/mixin/RosettaXMLAnnotationIntrospector.java

```diff
@@ -0,0 +1,266 @@
+import com.rosetta.util.serialisation.AttributeXMLRepresentation;
+import com.rosetta.util.serialisation.RosettaXMLConfiguration;
+import com.rosetta.util.serialisation.TypeXMLConfiguration;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class RosettaXMLAnnotationIntrospector extends JacksonXmlAnnotationIntrospector {
+
+    private static final long serialVersionUID = 1L;
+
+    private final RosettaXMLConfiguration rosettaXMLConfiguration;
+
+    private final RosettaEnumBuilderIntrospector rosettaEnumBuilderIntrospector;
+
+    private final EnumAsStringBuilderIntrospector enumAsStringBuilderIntrospector;
+
+
+    public RosettaXMLAnnotationIntrospector(final RosettaXMLConfiguration rosettaXMLConfiguration, final boolean supportNativeEnumValue) {
+        this(rosettaXMLConfiguration, new RosettaEnumBuilderIntrospector(supportNativeEnumValue), new EnumAsStringBuilderIntrospector());
+    }
+
+    public RosettaXMLAnnotationIntrospector(RosettaXMLConfiguration rosettaXMLConfiguration, RosettaEnumBuilderIntrospector rosettaEnumBuilderIntrospector, final EnumAsStringBuilderIntrospector enumAsStringBuilderIntrospector) {
+        this.rosettaXMLConfiguration = rosettaXMLConfiguration;
+        this.rosettaEnumBuilderIntrospector = rosettaEnumBuilderIntrospector;
+        this.enumAsStringBuilderIntrospector = enumAsStringBuilderIntrospector;
+    }
+
+    @Override
+    public Class<?> findPOJOBuilder(AnnotatedClass ac) {
+        if (ac.hasAnnotation(RosettaDataType.class)) {
+            return ac.getAnnotation(RosettaDataType.class).builder();
+        }
+        return super.findPOJOBuilder(ac);
+    }
+
+    @Override
+    public JsonPOJOBuilder.Value findPOJOBuilderConfig(AnnotatedClass ac)
+    {
+        if (ac.hasAnnotation(RosettaDataType.class)) {
+            // Our generated code uses 'build' for the build method, 'set' as setter prefix
+            return new JsonPOJOBuilder.Value("build", "set");
+        }
+        return super.findPOJOBuilderConfig(ac);
+    }
+
+    @Override
+    public PropertyName findRootName(AnnotatedClass ac) {
+        // If the root element name is specified in the XML configuration, use that.
+        // Otherwise, if the RosettaDataType annotation is present, use the name from that.
+        // Otherwise, use the default.
+        return getTypeXMLConfiguration(ac)
+                .flatMap(TypeXMLConfiguration::getXmlRootElementName)
+                .map(PropertyName::construct)
+                .orElseGet(() ->
+                        Optional.ofNullable(ac.getAnnotation(RosettaDataType.class))
+                                .map(rosettaDataTypeAnn -> PropertyName.construct(rosettaDataTypeAnn.value()))
+                                .orElseGet(() -> super.findRootName(ac))
+                );
+    }
+
+    @Override
+    protected PropertyName _findXmlName(Annotated a) {
```

Comment:
I think that we are not meant to override `_findXmlName` methods. Is that a caller method we can override?